### PR TITLE
Contract sources update

### PR DESCRIPTION
**Pull request by goastler**
Update the js code to the latest version of the contract

### CODE REVIEW

File: packages/provider/tests/tasks/tasks.test.ts [hunk 1]
```diff
@@ -321,27 +321,27 @@ describe('CONTRACT TASKS', async function (): Promise<void> {
...
+        const provider = (await tasks.contract.query.getProvider(accountAddress(providerAccount))).value
             .unwrap()
             .unwrap()
 
         const completedAt = (await env.api.rpc.chain.getBlock()).block.header.number.toNumber()
         const requestedAt = completedAt - 1
         const providerTasks = await getSignedTasks(env, providerAccount)
         await providerTasks.contract.tx.providerCommit({
-            dapp: accountContract(dappContractAccount),
+            dappContract: accountContract(dappContractAccount),
```
Comment (forgetso) [inline packages/provider/tests/tasks/tasks.test.ts:332]: Is there a reason why these have all changed back? I think you need to cd into provider and run `npm run build:test`

**Comment by goastler** [comment]
fixes #121

File: packages/types/src/contract/typechain/captcha/types-returns/captcha.ts [hunk 2]
```diff
@@ -34,34 +53,39 @@ export enum Error {
...
+export type UserHistorySummary = {
+	correct: number,
+	incorrect: number,
+	score: number
 }
 
 export type Commit = {
 	id: Hash,
-	user: AccountId,
+	userAccount: AccountId,
```
Comment (forgetso) [inline packages/types/src/contract/typechain/captcha/types-returns/captcha.ts:84]: Looks like we're both trying to accommodate for the other part. I had changed everything in TS to dapp/provider/user to match the contract.

**Review by forgetso — CHANGES_REQUESTED** [review]

File: packages/types/src/contract/typechain/captcha/types-returns/captcha.ts (hunk 2 above)
Comment (goastler) [reply packages/types/src/contract/typechain/captcha/types-returns/captcha.ts:84]: let's chat on slack :)

**Review by forgetso — DISMISSED** [review]

**Review by forgetso — APPROVED** [review]

**Pull request merged**